src/dtgtk/thumbnail.c

```diff
@@ -1098,26 +1098,30 @@ GtkWidget *dt_thumbnail_create_widget(dt_thumbnail_t *thumb)
       gtk_style_context_add_class(context, "dt_culling_thumb_image");
     gtk_widget_set_name(thumb->w_image, "thumb_image");
     // we pre-set the size of the imagebox with the aspect ratio so the image is already centered
-    const dt_image_t *img = dt_image_cache_get(darktable.image_cache, thumb->imgid, 'r');
     int iw = thumb->width;
     int ih = thumb->height;
-    if(img)
+    // we can't rely on img->aspect_ratio as the value is round to 1 decimal, so not enough accurate
+    // so we compute it from the larger available mipmap
+    for(int k = DT_MIPMAP_7; k >= DT_MIPMAP_0; k--)
     {
-      const float ar = img->aspect_ratio;
-      dt_image_cache_read_release(darktable.image_cache, img);
-      if(ar > 0.0001)
```

Comment:
Strange, my `ar` did always had more than 1 decimal, even before your change.

Reply:
Okay, that seems to depend on the image. The portrait I've used before had aspect ratio from image cache: 0.74545455, with your change it's 0.75 *and* I don't notice any flickering. Seems like you are on the right track, the correct aspect ratio is indeed 0.75 :tada: 

The portrait above has aspect ratio from image cache: 1.5
With your change, it is 1.50916672.
Once I've zoomed to 100% and re-open the image, the final_width and final_height get initialised. Then the aspect ratio is
When I take `(float)img->final_width / img->final_height` it is 1.50884688.

Conslusion: the mipmap cache improves the situation, but it is not always accurate enough.

Reply:
I figured out why it still had a flicker. It's a rounding problem!

I'll try to explain based on my sample image:
| value      | width | height |
| ----------- | ------- | ------ |
| thumb    | 1920 | 1139 |
| mipmap | 1811 | 1200 |
| initial thumb->w_image (in dt_thumbnail_create_widget) | 1920 | 1272 | 
| new thumb->w_image (in _thumb_set_image_area) |  1686.03784 | 1117 |
| cairo_image_surface_get_width(thumb->img_surf) (in _event_image_draw) | 1685 | 1117 |

The problem is in _thumb_set_image_area, where the width is scaled down when taking margins into account. I have in this example `image_w = 1882` and `image_h = 1117` and the initial w_image sizes. The resulting scale is
 ```const float scale = fminf((float)image_w / wi, (float)image_h / hi) = 0.878144681```.
The new image_w width is 1 px too large compared to what the final cairo calculated width returns.

Interestingly, when I revert your removal of the scaling (see https://github.com/darktable-org/darktable/pull/7719#discussion_r554220593), then I don't have any more flickering :tada: 

After adding the scaling, the initial width/height of thumb->w_image is 1719 x 1139.
That gives later in _thumb_set_image_area `scale = 0.980684817` and new thumb->w_image size 1685.79724 x 1117.

I suggest to re-add the scaling. The cairo surface is scaled in a similar way:
https://github.com/darktable-org/darktable/blob/master/src/views/view.c#L963-L967